src/java.base/share/classes/java/io/IO.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.io;
+
+import jdk.internal.javac.PreviewFeature;
+
+/**
+ * A collection of static convenience methods that provide access to
+ * {@link System#console()} for implicitly declared classes.
+ *
+ * <p> Each of this class' methods calls a similarly-named method on
+ * {@code Console} returned by {@code System.console()}, or throws
+ * {@link IOError} if {@code System.console()} returns {@code null}.
+ *
+ * @since 23
+ */
+@PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+public class IO {
+
+    private IO() {
+        throw new Error("no instances");
+    }
+
```

Comment:
For all three methods, the first sentence should state what the method does, as opposed to how it does it. This can (and probably should) be very similar to the corresponding methods on `Console`. For example, the first sentence of the`println` doc could be:

> Prints a string representation of the specified object to this console, followed by a newline.

And similar for the others.

Then, the "how" part should follow. I'd prefer to handle the null Console case first so that there's no confusion, then follow with "the effect is as if...":

> If System.console() returns null, throws IOError. Otherwise, the effect is as if
> System.console().println(obj) had been called.